c_tests/tests/compile_simple.c

```diff
@@ -13,13 +13,15 @@ int
 main(int argc, char **argv)
 {
     void *tt = __yktrace_start_tracing(HW_TRACING);
-    int res = argc + 1;
+    int res = 1 + 1;
```

Comment:
Do we compile these tests with optimisations? If so, this will probably be compiled to:

```
void *tt = __yktrace_start_tracing(HW_TRACING);
void *tr = __yktrace_stop_tracing(tt);
assert(2 == 2);
```

where the `assert(2==2)` will also be removed. So this trace will effectively be a no-op.

Reply:
When we checked the compiled code it still had a `mov reg, 2`, so it wasn't completely compiled away. So I guess optimisations are not on. I think Edd was experimenting with optimisations but hopefully by then we have more complicated tests that are not so easy to optimise away.

Reply:
My point is that the test really isn't robust -- the compiler can do whatever it wants to this code -- and if we don't fix it in this PR, we'll probably never fix it. Yes, it's annoying :) One way to do this is to have a function which either is (or looks like!) it's doing side effects being called between start/stop tracing. [We could use a trick along the lines of https://stackoverflow.com/questions/37786547/enforcing-statement-order-in-c/38025837#38025837 but there are other ways too.]

Reply:
We already had the idea to print to stdout and use lang tester to match it, but this was problematic as we can't yet do handle calls.

Reply:
Ah, we can't handle *any* function calls? If so, let's mark the test as "broken" or "fixme" and document why the test isn't robust.

Reply:
Yeah, I alluded a bit to this in the PR description. Something like 7c7aea5 will do?